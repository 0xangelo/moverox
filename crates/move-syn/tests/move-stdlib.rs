use move_syn::File;
use testresult::TestResult;
use unsynn::{IParse as _, ToTokens as _};

macro_rules! test_file {
    ($($file: ident),* $(,)?) => {$(
        #[test]
        fn $file() -> TestResult {
            let content = include_str!(concat!(
                "../../../move/move-stdlib/sources/",
                stringify!($file),
                ".move"
            ));
            let ast: File = content.to_token_iter().parse_all()?;
            insta::assert_snapshot!(ast.tokens_to_string());
            Ok(())
        }
    )*};
}

test_file!(
    address,
    ascii,
    bcs,
    bit_vector,
    bool,
    debug,
    fixed_point32,
    hash,
    macros,
    option,
    string,
    type_name,
    u128,
    u64,
    u8,
    unit_test,
    uq32_32,
    uq64_64,
    vector,
);
