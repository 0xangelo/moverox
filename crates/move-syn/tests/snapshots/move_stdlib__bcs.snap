---
source: crates/move-syn/tests/move-stdlib.rs
expression: ast.tokens_to_string()
---
# [doc = " Utility for converting a Move value to its binary representation in BCS (Binary Canonical"] # [doc = " Serialization). BCS is the binary encoding for Move resources and other non-module values"] # [doc = " published on-chain. See https://github.com/diem/bcs#binary-canonical-serialization-bcs for more"] # [doc = " details on BCS."] module std :: bcs ; # [doc = " Return the binary representation of `v` in BCS (Binary Canonical Serialization) format"] public native fun to_bytes < MoveValue > (v : & MoveValue) : vector < u8 > ;
